### PR TITLE
Fe/authentication fixes

### PR DESCRIPTION
**Pull request by ArnavNath550**
1. Hiding login button when user is already authenticated
2. Fixed some small problems in fetching user response 

## Summary by CodeRabbit

- **New Features**
	- Introduced a custom `ProductRouter` component for improved routing.
	- Added new tab content components for product details: About, Comments, and Team.
	- Launched a `ProductTechContent` component to display GitHub specifications and tech stack information.
- **Enhancements**
	- Streamlined user authentication process in the `AuthenticationModal`.
	- Improved code formatting and import order in the `SearchInput` component.
- **Documentation**
	- Added type definitions for authentication response.

### CODE REVIEW

File: apps/ui/src/components/modals/authentication/authentication-modal.tsx [hunk 1]
```diff
@@ -52,7 +52,7 @@ const AuthenticationModal: React.FC = () => {
 
     const handleUserSignIn = async (data: any) => {
```
Comment (shravan20) [inline apps/ui/src/components/modals/authentication/authentication-modal.tsx:53]: Can we create a type for the data?

**Review by shravan20 — APPROVED** [review]
PR LGTM; added minor comment

File: apps/ui/src/components/modals/authentication/authentication-modal.tsx [hunk 2]
```diff
@@ -52,7 +52,7 @@ const AuthenticationModal: React.FC = () => {
...
         const response = await signIn(data);
-        authLogin(response.data.user._id, response.data.access_token, response.data.user.email);
+        authLogin(response.payload.user._id, response.payload.access_token, response.payload.user.email);
         window.location.reload();
```
Comment (shravan20) [inline apps/ui/src/components/modals/authentication/authentication-modal.tsx:56]: Do we need a reload here? this would re-render whole screen?

**Review by shravan20 — APPROVED** [review]

**Review by shravan20 — CHANGES_REQUESTED** [review]
Hey. yet to be addressed?

**Comment by ArnavNath550** [comment]
@shravan20  done done:)
just fixed these ones up abhi

**Review by shravan20 — APPROVED** [review]
LGTM

**Pull request merged**